### PR TITLE
fix: assert :nth-child() selects children

### LINKED ISSUE #28544: "nth-child" is, surprisingly, applied to the child, not the parent
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child

### What specific section or headline is this issue about?

(unsure)

### What information was incorrect, unhelpful, or incomplete?

Even after I read this article several times, I still thought that this operator selected the nth child of a given parent. It is the obvious conclusion from the name "nth-child".

### What did you expect to see?

There needs to be a clarification that what this operator does is, given the child of a parent, selects (detects) whether that child is the nth child of the parent. Therefore, this operator operates on the child, not the parent node.

If you want a specific css rule to run when the nth child of a parent is hovered over, for example, you must write n separate statements, the first of which is `parent>child:nth-child(1):hover {}`. This example should appear in this article.

### Do you have any supporting links, references, or citations?

(No; I discovered this by discussing a time-consuming CSS programming error of mine with ChatGPT, then experimenting).

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/_colon_nth-child`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/_colon_nth-child/index.md
* Last commit: https://github.com/mdn/content/commit/764b291098cfe575374580844f1a91e7d814a8d3
* Document last modified: 2023-06-21T22:40:10.000Z

</details>

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- fixes https://github.com/mdn/content/issues/28544

### CODE REVIEW

File: files/en-us/web/css/_colon_nth-child/index.md [hunk 1]
```diff
@@ -7,7 +7,7 @@ browser-compat: css.selectors.nth-child
 
 {{CSSRef}}
 
-The **`:nth-child()`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) matches elements based on their position among a group of siblings.
+The **`:nth-child()`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) matches elements based on their position among a group of siblings. In other words, child elements are selected according to their position in a parent element.
```
Comment (estelle) [inline files/en-us/web/css/_colon_nth-child/index.md:10]:
```suggestion
The **`:nth-child()`** [CSS](/en-US/docs/Web/CSS) [pseudo-class](/en-US/docs/Web/CSS/Pseudo-classes) matches elements based on the indexes of the elements in the child list of their parents. In other words, the `:nth-child()` selector selects child elements according to their position among all the sibling elements within a parent element.
```

the "all" is really important, because "type" counts only of a specific type but child considers all the children equally. We  might want to and the word "all" to line 14 as well.

**Review by estelle — COMMENTED** [review]
i added a few see alsos and tried to clarify the intro sentence further.

File: files/en-us/web/css/_colon_nth-child/index.md [hunk 2]
```diff
@@ -424,3 +424,6 @@ In the second table the _of syntax_ is used to target only the `tr`s that are **
 
 - {{ Cssxref(":nth-of-type") }}
 - {{ Cssxref(":nth-last-child") }}
+- {{ Cssxre(":has()") }}
```
Comment (OnkarRuikar) [inline files/en-us/web/css/_colon_nth-child/index.md:427]: Adhering to [See also section guidelines](https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Writing_style_guide#descriptive_text):
```suggestion
- {{ Cssxre(":has()") }}: pseudo-class for selecting parent element
```

**Review by estelle — APPROVED** [review]

**Comment by estelle** [comment]
Thank you!

File: files/en-us/web/css/_colon_nth-child/index.md [hunk 3]
```diff
@@ -424,3 +424,6 @@ In the second table the _of syntax_ is used to target only the `tr`s that are **
 
 - {{ Cssxref(":nth-of-type") }}
 - {{ Cssxref(":nth-last-child") }}
+- {{ Cssxre(":has()") }}: pseudo-class for selecting parent element
+- [Tree structural pseudo-classes](/en-US/docs/Web/CSS/Pseudo-classes#tree-structural_pseudo-classes
```
Comment (Josh-Cena) [inline files/en-us/web/css/_colon_nth-child/index.md:428]: Missing a `)`?

Comment (OnkarRuikar) [reply files/en-us/web/css/_colon_nth-child/index.md:428]: How did you find it?! I remember you correcting missing something like `)` before.

Looks like markdownlint can not detect it. Will investigate tomorrow. May be we need a new custom rule for this.

Comment (Josh-Cena) [reply files/en-us/web/css/_colon_nth-child/index.md:428]: I just randomly took a look at this commit and found no syntax highlighting here:)

A custom rule is doable, I think.

Comment (OnkarRuikar) [reply files/en-us/web/css/_colon_nth-child/index.md:428]: For now https://github.com/mdn/content/pull/28706

Comment (estelle) [reply files/en-us/web/css/_colon_nth-child/index.md:428]: it's missing the "F"

**Pull request merged**